src/components/NoteCardList.jsx

```diff
@@ -1,31 +1,47 @@
-import { useContext } from 'react';
+import { useContext, useState, useEffect } from 'react';
 import { GlobalContext } from '../context/GlobalContext';
 import { CardListContainer } from './styles/CardListContainer.styled';
 import NoteCard from './NoteCard';
+import NoNotesAlert from './NoNotesAlert';
 import PropTypes from 'prop-types';
 
-const NoteCardList = ({ archivesPage }) => {
-  const { notes, archives } = useContext(GlobalContext);
-  let iterator = notes;
-  if (archivesPage) {
-    iterator = archives;
-  }
+const NoteCardList = ({ archivesPage, handleShowEditModal }) => {
+  const { notes, archives, query } = useContext(GlobalContext);
+  const [filteredNotes, setFilteredNotes] = useState([]);
+
+  useEffect(() => {
+    setFilteredNotes(
+      notes.filter((note) =>
+        note.noteText.toLowerCase().includes(query.toLowerCase())
+      )
+    );
+  }, [query, notes]);
+
+  const filteredNotesOrArchivesArray = archivesPage ? archives : filteredNotes;
```

Comment:
i think a better name for this is notesList or filteredNotes (just a suggestion as it doesn't affect anything